ui/src/app/views/workspace-nav-bar/component.tsx

```diff
@@ -0,0 +1,150 @@
+    height: 60, paddingRight: 16,
+    boxShadow: 'inset rgba(0, 0, 0, 0.12) 0px 3px 2px 0px',
+    width: 'calc(100% + 1.2rem)',
+    marginLeft: '-0.6rem',
+    paddingLeft: 80, borderBottom: `5px solid ${colors.blue[0]}`, flex: 'none'
+  },
+  tab: {
+    minWidth: 140, flexGrow: 0, padding: '0 20px',
+    color: colors.gray[4],
+    alignSelf: 'stretch', display: 'flex', justifyContent: 'center', alignItems: 'center'
+  },
+  active: {
+    backgroundColor: 'rgba(255,255,255,0.15)', color: 'unset',
+    borderBottom: `4px solid ${colors.blue[0]}`, fontWeight: 'bold'
+  },
+  separator: {
+    background: 'rgba(255,255,255,0.15)', width: 1, height: 48, flexShrink: 0
+  },
+  dropdownHeader: {
+    fontSize: 12,
+    lineHeight: '30px',
+    color: '#262262',
+    fontWeight: 600,
+    paddingLeft: 12,
+    width: 160
+  },
+  menuButtonIcon: {
+    width: 27, height: 27,
+    opacity: 0.65, marginRight: 16
+  }
+});
+
+const tabs = [
+  {name: 'About', link: ''},
+  {name: 'Cohorts', link: 'cohorts'},
+  {name: 'Concepts', link: 'concepts'},
+  {name: 'Notebooks', link: 'notebooks'}
+];
+
+const navSeparator = <div style={styles.separator}/>;
+
+export const WorkspaceNavBarReact = withCurrentWorkspace()(props => {
+  const {shareFunction, deleteFunction, workspace, tabPath} = props;
+  const {namespace, id, accessLevel} = workspace;
+  const isNotOwner = accessLevel !== WorkspaceAccessLevel.OWNER;
+  const activeTabIndex = fp.findIndex(['link', tabPath], tabs);
+
+
+  const navTab = currentTab => {
+    const {name, link} = currentTab;
+    const selected = tabPath === link;
+    const hideSeparator = selected || (activeTabIndex === tabs.indexOf(currentTab) + 1);
+
+    return <React.Fragment key={name}>
+      <Clickable
+        data-test-id={name}
+        aria-selected={selected}
+        style={{...styles.tab, ...(selected ? styles.active : {})}}
+        hover={{color: styles.active.color}}
+        onClick={() => NavStore.navigate(fp.compact(['/workspaces', namespace, id, link]))}
+      >
+        {name}
+      </Clickable>
+      {!hideSeparator && navSeparator}
+    </React.Fragment>;
+  };
+
+  return <div id='workspace-top-nav-bar' className='do-not-print' style={styles.container}>
+    {activeTabIndex > 0 && navSeparator}
+    {fp.map(tab => navTab(tab), tabs)}
+    <div style={{flexGrow: 1}}/>
+    <PopupTrigger
+      side='bottom'
+      closeOnClick={true}
+      content={
+        <React.Fragment>
+          <div style={styles.dropdownHeader}>Workspace Actions</div>
+          <MenuItem
+            icon='copy'
+            onClick={() => NavStore.navigate(['/workspaces', namespace, id, 'clone'])}>
+            Clone
+          </MenuItem>
+          <MenuItem
+            icon='pencil'
+            tooltip={isNotOwner && 'Requires owner permission'}
+            disabled={isNotOwner}
+            onClick={() => NavStore.navigate(['/workspaces', namespace, id, 'edit'])}
+          >
```

Comment:
nit - put this `>` on above line

Reply:
I've been doing this everywhere so that there's a line of visibly different indentation between the props of a component and its children, so that you can tell where one ends and the other begins. If we really hate it I can change it, but I find it very helpful for navigating code visually